hbase-server/src/main/resources/hbase-webapps/master/hbck.jsp

```diff
@@ -0,0 +1,139 @@
+    orphanRegionsOnFS = hbckChecker.getOrphanRegionsOnFS();
+  }
+%>
+<jsp:include page="header.jsp">
+  <jsp:param name="pageTitle" value="${pageTitle}"/>
+</jsp:include>
+
+<div class="container-fluid content">
+
+  <% if (!master.isInitialized()) { %>
+  <div class="row inner_header">
+    <div class="page-header">
+      <h1>Master is not initialized</h1>
+    </div>
+  </div>
+  <jsp:include page="redirect.jsp" />
+  <% } else { %>
+
+  <div class="row inner_header">
+    <div class="page-header">
+      <h1>Inconsistent Regions</h1>
+      <p>
+        <span>
+        There are three case: 1. Master thought this region opened, but no regionserver reported it.
+        2. Master thought this region opened on Server1, but regionserver reported Server2.
+        3. More than one regionservers reported opened this region.
+        Notice: the reported online regionservers may be not right when there are regions in transition.
+        Please check them in regionserver's web UI.
+        </span>
+      </p>
+    </div>
+  </div>
+
+  <% if (inconsistentRegions != null && inconsistentRegions.size() > 0) { %>
+  <table class="table table-striped">
+    <tr>
+      <th>Region</th>
+      <th>Location in META</th>
+      <th>Reported Online RegionServers</th>
+    </tr>
+    <% for (Map.Entry<String, Pair<ServerName, List<ServerName>>> entry : inconsistentRegions.entrySet()) { %>
+    <tr>
+      <td><%= entry.getKey() %></td>
+      <td><%= entry.getValue().getFirst() %></td>
+      <td><%= entry.getValue().getSecond().stream().map(ServerName::getServerName)
+                        .collect(Collectors.joining(", ")) %></td>
+    </tr>
+    <% } %>
+
+    <p><%= inconsistentRegions.size() %> region(s) in set.</p>
+  </table>
+  <% } %>
+
+  <div class="row inner_header">
+    <div class="page-header">
+      <h1>Orphan Regions on RegionServer</h1>
+    </div>
+  </div>
+
+  <% if (orphanRegionsOnRS != null && orphanRegionsOnRS.size() > 0) { %>
+  <table class="table table-striped">
+    <tr>
+      <th>Region</th>
+      <th>Reported Online RegionServer</th>
+    </tr>
+    <% for (Map.Entry<String, ServerName> entry : orphanRegionsOnRS.entrySet()) { %>
+    <tr>
+      <td><%= entry.getKey() %></td>
+      <td><%= entry.getValue() %></td>
+    </tr>
+    <% } %>
+
+    <p><%= orphanRegionsOnRS.size() %> region(s) in set.</p>
+  </table>
+  <% } %>
+
+  <div class="row inner_header">
+    <div class="page-header">
+      <h1>Orphan Regions on FileSystem</h1>
+    </div>
+  </div>
+
+  <% if (orphanRegionsOnFS != null && orphanRegionsOnFS.size() > 0) { %>
+  <table class="table table-striped">
+    <tr>
+      <th>Region</th>
+    </tr>
+    <% for (String region : orphanRegionsOnFS) { %>
+    <tr>
+      <td><%= region %></td>
+    </tr>
+    <% } %>
+
+    <p><%= orphanRegionsOnFS.size() %> region(s) in set.</p>
+  </table>
+  <% } %>
+
+  <% } %>
+</div>
+
+<jsp:include page="footer.jsp"/>
```

Comment:
Beautiful.

In follow-on, we can add text on how to fix including commands.

Also may want a json version of this page..... Lets see.

Reply:
Ok.

Reply:
> Beautiful.
> 
> In follow-on, we can add text on how to fix including commands.
> 
> Also may want a json version of this page..... Lets see.

Let me open a new issue for this.